src/server/status_server.rs

```diff
@@ -41,26 +87,131 @@ impl StatusServer {
+            Ok(lock) => lock,
+        };
+        info!("start memory profiling {} seconds", seconds);
+
+        let timer = GLOBAL_TIMER_HANDLE.clone();
+        Box::new(lock.then(move |guard| {
+            timer
+                .delay(std::time::Instant::now() + std::time::Duration::from_secs(seconds))
+                .then(
+                    move |_| -> Box<dyn Future<Item = Vec<u8>, Error = ProfError> + Send> {
+                        let tmp_dir = match TempDir::new("") {
+                            Ok(tmp_dir) => tmp_dir,
+                            Err(e) => return Box::new(err(e.into())),
+                        };
+                        let os_path = tmp_dir.path().join("tikv_dump_profile").into_os_string();
+                        let path = match os_path.into_string() {
+                            Ok(path) => path,
+                            Err(path) => return Box::new(err(ProfError::PathError(path))),
+                        };
+
+                        if let Err(e) = tikv_alloc::dump_prof(&path) {
+                            return Box::new(err(e));
+                        }
+                        drop(guard);
+                        Box::new(
+                            tokio_fs::file::File::open(path)
+                                .and_then(|file| {
+                                    let buf: Vec<u8> = Vec::new();
+                                    tokio_io::io::read_to_end(file, buf)
+                                })
+                                .and_then(move |(_, buf)| {
+                                    drop(tmp_dir);
+                                    ok(buf)
+                                })
+                                .map_err(|e| -> ProfError { e.into() }),
+                        )
+                    },
+                )
+        }))
+    }
+
+    pub fn dump_prof_to_resp(
+        req: Request<Body>,
+    ) -> Box<dyn Future<Item = Response<Body>, Error = hyper::Error> + Send> {
+        let query = match req.uri().query() {
+            Some(query) => query,
+            None => {
+                let response = Response::builder()
+                    .status(StatusCode::BAD_REQUEST)
+                    .body(Body::empty())
+                    .unwrap();
+                return Box::new(ok(response));
+            }
+        };
+        let query_pairs: HashMap<_, _> = url::form_urlencoded::parse(query.as_bytes()).collect();
+        let seconds: u64 = match query_pairs.get("seconds") {
+            Some(val) => match val.parse() {
+                Ok(val) => val,
+                Err(_) => {
+                    let response = Response::builder()
+                        .status(StatusCode::BAD_REQUEST)
+                        .body(Body::empty())
+                        .unwrap();
+                    return Box::new(ok(response));
+                }
+            },
+            None => {
+                let response = Response::builder()
```

Comment:
Can it be set to a default value like 60 seconds instead? What's the behavior of TiDB?

Reply:
```
sec, err := strconv.ParseInt(r.FormValue("seconds"), 10, 64)
if sec <= 0 || err != nil {
    sec = 10
} 
```
The default value in TiDB is 10 seconds.

Reply:
Now,  if `seconds` query parameter is not passed, the value will be set to 10 seconds. If it's not a integer, an error will be returned.